aea/registries/base.py

```diff
@@ -46,43 +47,50 @@
 PACKAGE_NAME_REGEX = re.compile(
     "^([A-Z0-9]|[A-Z0-9][A-Z0-9._-]*[A-Z0-9])$", re.IGNORECASE
 )
-INTERNAL_PROTOCOL_ID = "internal"
+INTERNAL_PROTOCOL_ID = PublicId.from_string("fetchai/internal:0.1.0")
 DECISION_MAKER = "decision_maker"
 
 
-class Registry(ABC):
+Item = TypeVar("Item")
+ItemId = TypeVar("ItemId")
+ComponentId = Tuple[SkillId, str]
+SkillComponentType = TypeVar("SkillComponentType", Handler, Behaviour, Task)
+
+
+class Registry(Generic[ItemId, Item], ABC):
```

Comment:
a `Registry` is a generic type. The user of this class (and subclasses) "must" specify (read "must" as "should", since Python is not very strict)  what is the `ItemId` type used to reference the items, and the `Item` type.

Reply:
@DavidMinarsch  We do! e.g. the tuple `public_id, name` for skill components